roles/junit2json/tasks/convert.yml

```diff
@@ -0,0 +1,100 @@
+    junit2json_xml_report_content: "{{ lookup('ansible.builtin.file', xml_report) }}"
+
+- name: Generate content hash from the content
+  ansible.builtin.set_fact:
+    junit2json_xml_report_hash_path: "{{ xml_report }}.{{ junit2json_hash }}"
+    junit2json_xml_report_hash_curr: "{{ junit2json_xml_report_content | ansible.builtin.hash(junit2json_hash) }}"
+    junit2json_xml_report_hash_old: ""
+
+# debug only:
+- name: Print curr XML report filename
+  ansible.builtin.debug:
+    msg:
+      - "XML report path is: {{ xml_report }}"
+  when:
+    - junit2json_debug
+
+- name: Obtain info on previous content checksum file
+  ansible.builtin.stat:
+    path: "{{ junit2json_xml_report_hash_path }}"
+    checksum_algorithm: "{{ junit2json_hash }}"
+  register: _junit2json_old_hash_stat
+
+- name: "Update junit2json_result_data [`junit2json_do_merge: true`]"
+  ansible.builtin.set_fact:
+    junit2json_result_data: "{{ _junit2json_merged_xml.stdout | redhatci.ocp.junit2obj | to_nice_json }}"
+  when:
+    - junit2json_do_merge
+
+- name: Convert junit XML to JSON and save in junit2json_result_data
+  ansible.builtin.set_fact:
+    junit2json_result_data: "{{ junit2json_xml_report_content | redhatci.ocp.junit2obj | to_nice_json }}"
+  when:
+    - not(junit2json_do_merge)
+
+- name: "Setup JSON report file name [`xml_report.endswith('.xml'): true`]"
+  ansible.builtin.set_fact:
+    junit2json_json_report_path: "{{ xml_report | basename | regex_replace('\\.xml$', '.json') }}"
+  when:
+    - not(junit2json_do_merge)
+    - xml_report.endswith(".xml")
+
+- name: "Setup JSON report file name [`xml_report.endswith('.xml'): false`]"
+  ansible.builtin.set_fact:
+    junit2json_json_report_path: "{{ xml_report | basename }}.json"
+  when:
+    - not(junit2json_do_merge)
+    - not(xml_report.endswith(".xml"))
+
+- name: "Update junit2json_output_report_path [`junit2json_do_merge: true`]"
+  ansible.builtin.set_fact:
+    junit2json_output_report_path: "{{ junit2json_output_dir }}/{{ junit2json_output_merged_report }}"
+  when:
+    - junit2json_do_merge
+
+- name: "Update junit2json_output_report_path [`junit2json_do_merge: false`]"
+  ansible.builtin.set_fact:
+    junit2json_output_report_path: "{{ junit2json_output_dir }}/{{ junit2json_json_report_path }}"
+  when:
+    - not(junit2json_do_merge)
+
+- name: Get xml_old_hash
+  ansible.builtin.set_fact:
+    junit2json_xml_report_hash_old: "{{ _junit2json_old_hash_stat.stat.checksum }}"
+  when:
+    - _junit2json_old_hash_stat.stat.exists
+
+- name: Update global_json_reports_list
+  ansible.builtin.set_fact:
+    global_json_reports_list: "{{ global_json_reports_list | default([]) + [junit2json_output_report_path] }}"
+
+- name: Ensure json data destination folder
+  ansible.builtin.file:
+    path: "{{ junit2json_output_dir }}"
+    state: directory
+    mode: '0755'
+  when:
+    - junit2json_xml_report_hash_curr != junit2json_xml_report_hash_old
+
+- name: Write json data to destination
```

Comment:
For the following two Write tasks, you can use a block with a single when conditional, as they share the same logic and content approx.

Reply:
Please reply before resolving.